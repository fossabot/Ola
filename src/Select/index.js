import React, {forwardRef} from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const SelectOption = ({ value, label, ...props }) =>  <option value={value} {...props}>{label}</option>

const Select = forwardRef(({ options, error, className, ...props }, ref) => {
  const styles = cx('ola_select', {'is-invalid': error}, className)
  return (
    <select className={styles} ref={ref} {...props}>
      { options.map( ({ value, label }, idx) => <SelectOption key={idx} value={value} label={label} /> ) }
    </select>
  )
})

Select.defaultProps = {
  className: null,
}

Select.displayName = 'Select'

Select.propTypes = {
  /** Select options (array {value: any, label: string} )*/
  options: PT.arrayOf(PT.shape({
    value: PT.any,
    label: PT.string
  })).isRequired,
  /** Extra className */
  className: PT.string,
  /** Select is invalid */
  error: PT.bool
}

export default Select
