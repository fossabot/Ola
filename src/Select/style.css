.ola_select {
    --border: var(--gray-light);
    --color: var(--black);
    --background: var(--white);
    --focus: var(--brand-focus);

    display: block;
    appearance: none;
    box-sizing: border-box;
    height: var(--size-8);
    width: 100%;
    border: solid 2px var(--border);
    background: var(--white);
    color: var(--color);
    background-color: var(--background);
    /* Chevron gray-light */
    background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="44px" height="44px" viewBox="0 0 44 44" version="1.1" xmlns="http://www.w3.org/2000/svg"><path d="M22,23.5857864 L27.2928932,18.2928932 C27.6834175,17.9023689 28.3165825,17.9023689 28.7071068,18.2928932 C29.0976311,18.6834175 29.0976311,19.3165825 28.7071068,19.7071068 L22.7071068,25.7071068 C22.3165825,26.0976311 21.6834175,26.0976311 21.2928932,25.7071068 L15.2928932,19.7071068 C14.9023689,19.3165825 14.9023689,18.6834175 15.2928932,18.2928932 C15.6834175,17.9023689 16.3165825,17.9023689 16.7071068,18.2928932 L22,23.5857864 Z" fill="%23BBC2CE"></path></svg>');
    background-repeat: no-repeat;
    background-position: right center;
    border-radius: var(--radius-small);
    font: var(--font-callout);
    padding: 0 var(--size-8) 0 1em;
    transition-property: background-color, color, box-shadow, border-color;
    transition-duration: 200ms;
    outline: 0;
    white-space: nowrap;
    text-decoration: none;
    font-weight: var(--regular);

    &:focus {
        --border: var(--brand);
        box-shadow: 0 0 0 4px var(--focus);

        /* Chevron brand */
        background-image: url('data:image/svg+xml;charset=UTF-8,<svg width="44px" height="44px" viewBox="0 0 44 44" version="1.1" xmlns="http://www.w3.org/2000/svg"><path d="M22,23.5857864 L27.2928932,18.2928932 C27.6834175,17.9023689 28.3165825,17.9023689 28.7071068,18.2928932 C29.0976311,18.6834175 29.0976311,19.3165825 28.7071068,19.7071068 L22.7071068,25.7071068 C22.3165825,26.0976311 21.6834175,26.0976311 21.2928932,25.7071068 L15.2928932,19.7071068 C14.9023689,19.3165825 14.9023689,18.6834175 15.2928932,18.2928932 C15.6834175,17.9023689 16.3165825,17.9023689 16.7071068,18.2928932 L22,23.5857864 Z" fill="%230090DA"></path></svg>');
    }

    &:invalid,
    &.is-invalid {
        --border: var(--error);
        --color: var(--error);
        box-shadow: none;

        &:focus {
            box-shadow: 0 0 0 4px var(--error-focus);
        }
    }

    &:disabled {
        --border: var(--gray-xlight);
        --color: var(--gray-light);
        cursor: not-allowed;

        &::placeholder {
            opacity: 0;
        }
    }
}
