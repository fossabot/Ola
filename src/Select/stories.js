import React from 'react'
import Select from './'

export default {
  title: 'Select',
  component: Select,
  args: {
    options: [
      { value: '', label: 'Iâ€™m not sure' },
      { value: 'chocolate', label: 'Chocolate' },
      { value: 'strawberry', label: 'Strawberry' },
      { value: 'vanilla', label: 'Vanilla' },
      { value: '', label: 'None' }
    ]
  },
  argTypes: {
    onClick: { action: 'clicked' },
    onChange: { action: 'changed' },
  }
}

export const Base = (args) => <Select {...args}/>
