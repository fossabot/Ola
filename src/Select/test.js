import React from 'react'
import renderer from 'react-test-renderer'

import Select from './'

const testOptions = [
  { value: 'chocolate', label: 'Chocolate' },
  { value: 'strawberry', label: 'Strawberry' },
  { value: 'vanilla', label: 'Vanilla' },
  { value: '', label: 'None' }
]

it('Default Page', () => {
  const tree = renderer
    .create(<Select options={testOptions} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Default defaultValue', () => {
  const tree = renderer
    .create(<Select options={testOptions} defaultValue="strawberry" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Default with value', () => {
  const tree = renderer
    .create(<Select options={testOptions} value="vanilla" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('Default with empty option', () => {
  const tree = renderer
    .create(<Select options={testOptions} emptyOption="Empty option" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
