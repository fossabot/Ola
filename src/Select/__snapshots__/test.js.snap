// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Page 1`] = `
<select
  className="ola_select"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value=""
  >
    None
  </option>
</select>
`;

exports[`Default defaultValue 1`] = `
<select
  className="ola_select"
  defaultValue="strawberry"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value=""
  >
    None
  </option>
</select>
`;

exports[`Default with value 1`] = `
<select
  className="ola_select"
  value="vanilla"
>
  <option
    value="chocolate"
  >
    Chocolate
  </option>
  <option
    value="strawberry"
  >
    Strawberry
  </option>
  <option
    value="vanilla"
  >
    Vanilla
  </option>
  <option
    value=""
  >
    None
  </option>
</select>
`;
